src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -10939,26 +10944,52 @@ def _window_agg(
         # Handle case where min_periods = None
         min_periods = 0 if min_periods is None else min_periods
         # Perform Aggregation over the window_expr
-        new_frame = frame.update_snowflake_quoted_identifiers_with_expressions(
-            {
-                # If aggregation is count use count on row_position_quoted_identifier
-                # to include NULL values for min_periods comparison
-                quoted_identifier: iff(
-                    count(col(row_position_quoted_identifier)).over(window_expr)
-                    >= min_periods
-                    if agg_func == "count"
-                    else count(col(quoted_identifier)).over(window_expr) >= min_periods,
-                    get_snowflake_agg_func(agg_func, agg_kwargs)(
-                        # Expanding is cumulative so replace NULL with 0 for sum aggregation
-                        builtin("zeroifnull")(col(quoted_identifier))
-                        if window_func == WindowFunction.EXPANDING and agg_func == "sum"
-                        else col(quoted_identifier)
-                    ).over(window_expr),
-                    pandas_lit(None),
-                )
-                for quoted_identifier in frame.data_column_snowflake_quoted_identifiers
-            }
-        ).frame
+        if agg_func == "sem":
+            # Standard error of mean (SEM) does not have native Snowflake engine support
+            # so calculate as STDDEV/SQRT(N-1)
+            new_frame = frame.update_snowflake_quoted_identifiers_with_expressions(
+                {
+                    quoted_identifier: iff(
+                        count(col(quoted_identifier)).over(window_expr) >= min_periods,
+                        # If (N-1) is negative number, return NaN instead of attempting to sqrt
```

Comment:
If I'm understanding correctly, N-1 < 0 would only occur if a window has no elements? Is returning NaN the expected behavior here?

Also, what happens when N-1=0? Do we just propagate whatever error Snowflake raises for division by 0?

Reply:
Also, if you want to share logic for this with other aggregations, you could add it as a lambda in aggregation_utils.py.

https://github.com/snowflakedb/snowpark-python/blob/c91cca88cb1bbf41a1441877ecfcc9d60d0c474d/src/snowflake/snowpark/modin/plugin/_internal/aggregation_utils.py#L239

Reply:
@sfc-gh-joshi I added `test_expanding_sem_ddof` that tests different ddof values as well as correspondingly when `SQRT(N-ddof)` is values 1, 0, -0.5, -1.

I added tests for different ddof values for rolling, but testing the SQRT there is much harder since we don't yet support `window=0, min_periods=0` in rolling.